x-pack/plugins/alerting/public/alert_navigation_registry/alert_navigation_registry.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import Boom from 'boom';
+import { i18n } from '@kbn/i18n';
+import { AlertType } from '../../common';
+import { AlertNavigationHandler } from './types';
+
+const DEFAULT_HANDLER = Symbol('*');
+export class AlertNavigationRegistry {
+  private readonly alertNavigations: Map<
+    string,
+    Map<string | symbol, AlertNavigationHandler>
+  > = new Map();
+
+  public has(consumer: string, alertType: AlertType) {
+    return this.hasTypedHandler(consumer, alertType) || this.hasDefaultHandler(consumer);
+  }
+
+  public hasTypedHandler(consumer: string, alertType: AlertType) {
+    return this.alertNavigations.get(consumer)?.has(alertType.id) ?? false;
+  }
+
+  public hasDefaultHandler(consumer: string) {
+    return this.alertNavigations.get(consumer)?.has(DEFAULT_HANDLER) ?? false;
+  }
+
+  private createConsumerNavigation(consumer: string) {
+    const consumerNavigations = new Map<string, AlertNavigationHandler>();
+    this.alertNavigations.set(consumer, consumerNavigations);
+    return consumerNavigations;
+  }
+
+  public registerDefault(consumer: string, handler: AlertNavigationHandler) {
+    if (this.hasDefaultHandler(consumer)) {
+      throw Boom.badRequest(
```

Comment:
I guess I never noticed before that you can use `Boom` in ui plugins!  I'm guessing there must be some nice handling for these somewhere.  Just wanted to note this, to make sure using Boom is kosher, compared to I guess just throwing a normal exception.